samples/rust/src/walletcore.rs

```diff
@@ -0,0 +1,69 @@
+use libc::c_char;
+use std::ffi::CString;
+use std::ffi::CStr;
+
+// signatures
+extern "C" {
+    fn TWStringCreateWithUTF8Bytes(twstring: *const c_char) -> *const i8;
+    fn TWStringUTF8Bytes(twstring: *const i8) -> *const c_char;
+
+    fn TWHDWalletCreateWithMnemonic(mnemonic: *const i8, passphrase: *const i8) -> *const i8;
+    fn TWHDWalletGetAddressForCoin(wallet: *const i8, coin: i32) -> *const i8;
+
+    fn TWMnemonicIsValid(mnemonic: *const i8) -> bool;
+}
+
+// type utilities
+pub struct TWString {
+    wrapped: *const i8
+    // TODO delete when destructing with TWStringDelete
+}
+
+pub trait FromString {
+    fn from_str(s: &str) -> Self;
+}
+
+impl FromString for TWString {
+    //fn from_str(s: &str) -> Result<Self, Self::Err> {
+    fn from_str(s: &str) -> Self {
+        let cstring = CString::new(s).unwrap();
+        let ptr = unsafe { TWStringCreateWithUTF8Bytes(cstring.as_ptr()) };
+        return TWString { wrapped: ptr };
+    }
+}
+
+impl ToString for TWString {
+    fn to_string(&self) -> String {
+        let s1 = unsafe { TWStringUTF8Bytes(self.wrapped) };
+        let c_str: &CStr = unsafe { CStr::from_ptr(s1) };
+        let str_slice: &str = c_str.to_str().unwrap();
```

Comment:
Better to use a `Result` and `?` operator or rename the function `to_string_unsafe`

Note in rust you can implement the traits `From` `TryFrom` for such operation: https://doc.rust-lang.org/stable/std/convert/trait.TryFrom.html

Reply:
I skip this in this sample